src/Transformer/DateTimeIso8601Transformer.php

```diff
@@ -14,15 +14,16 @@ class DateTimeIso8601Transformer extends AbstractTransformer
     public function transform(ResourceInterface $resource)
     {
         $value = $this->getPropertyValueFromResource($resource);
+        if (empty($value)) {
```

Comment:
out of curiosity, why was this added? when passing into setters, if it's null, then a "typed" param is allowed. otherwise, if it's an '', a type error is returned.


`setDate(Carbon $date = null)`
if untransformed to `null`, this will work. if untransformed to `''`, this will fail